includes/RestApi/PluginsController.php

```diff
@@ -242,4 +278,59 @@ public function get_status( \WP_REST_Request $request ) {
 		);
 
 	}
+
+	/**
+	  * Retrieves the Customized list of Plugins for the user.
+	  *
+	  * @return array|\WP_Error
+	  */
+	public function get_custom_plugins() {
+		 $custom_plugins = SiteFeatures::get_custom_plugins_list();
+		 return $custom_plugins;
+	}
+
+	/**
+	 * Installs/Uninstalls the requested plugins.
+	 *
+	 * @param \WP_REST_Request $request
+	 *
+	 * @return \WP_REST_Response|\WP_Error
+	 */
+	public function uninstall( \WP_REST_Request $request ) {
+
+		  $plugin_body = json_decode( $request->get_body(), true );
+		  $plugins     = isset( $plugin_body['plugins'] ) ? $plugin_body['plugins'] : false;
+
+		if ( $plugins ) {
+			foreach ( $plugins as $plugin => $decision ) {
+				if ( $decision ) {
```

Comment:
Using the decision we need to either Install or Uninstall so the logic for that should be in a much more generic place and this logic was very much specific to the site features POST request.

Reply:
An API named `uninstall` is queuing plugins for install right now.

So the architecture I was thinking of is something like this
2. We can do `PluginUninstallTaskManager::uninstall()` in the `uninstall` API, the TaskManager's uninstall can decide whether to un-queue or uninstall. If not, I am also fine with keeping the logic in the API as well.
3. Point 2 is the only thing the `uninstall` API should be doing.
4. Since we already have a `GET site-features` endpoint we can create one more for handling `site-features` which can make the `decision` of calling the `install` or the `uninstall` API's internally?

We create a new story to handle the `long-term` approach for `site-features` and `plugin_init` mappings based on flow/plan.